homeassistant/components/wsdot/sensor.py

```diff
@@ -137,19 +142,24 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
                 ATTR_TRAVEL_TIME_ID,
             ):
                 attrs[key] = self._data.get(key)
-            attrs[ATTR_TIME_UPDATED] = _parse_wsdot_timestamp(
-                self._data.get(ATTR_TIME_UPDATED)
-            )
+            time_updated = self._data.get(ATTR_TIME_UPDATED)
+            if isinstance(time_updated, str):
+                attrs[ATTR_TIME_UPDATED] = _parse_wsdot_timestamp(time_updated)
+            else:
+                attrs[ATTR_TIME_UPDATED] = time_updated
```

Comment:
Can this type change?

Reply:
Which type, `ATTR_TIME_UPDATED`? I'm not sure what would make more sense than a datetime.

Reply:
I get that, but this wasn't here before so I'm wondering why it is now, if this is because of typing, we need to check why, if this is fixing a bug it should be its own PR

Reply:
Oh yes, this is because I typed `_data` and didn't use a TypedDict, so before this call site was unchecked, but now mypy knows that data's values are one of `str | int | float | datetime`

Reply:
okay, but what is the type of the source data?

Reply:
The value returned by the GET request can be either a string or null for this particular field.

Reply:
right, but then this should be a check for None and not something that looks like it can be different